drivers/fuel_gauge/lc709203f/emul_lc709203f.c

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (c) 2025 Philipp Steiner <philipp.steiner1987@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Emulator for lc709203f fuel gauge
+ */
+
+#include <string.h>
+#define DT_DRV_COMPAT onnn_lc709203f
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(EMUL_LC709203F);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/i2c_emul.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "lc709203f.h"
+
+/* You can store as many registers as you need.
+ * Note: The LC709203F typically uses 16-bit registers.
+ */
+struct lc709203f_emul_data {
+	/* This emulator object (required for i2c_emul). */
+	struct i2c_emul emul;
+	/* The I2C emulation config (pointer to our dev_config). */
+	const struct i2c_emul_api *api;
+	/* A backing store for registers in the device. */
+	uint16_t regs[0x1B]; /* or enough to hold all used registers */
+};
+
+struct lc709203f_emul_cfg {
+	/** I2C address of emulator */
+	uint16_t addr;
+};
+
+#define LC709203F_CRC_POLYNOMIAL 0x07 // Polynomial to calculate CRC-8-ATM
+
+/*
+ * Calculate the CRC-8 checksum using polynomial 0x07 (CRC-8-ATM).
+ */
+static uint8_t lc709203f_calc_crc(uint8_t *data, size_t data_len)
+{
+	uint8_t crc = 0;
+
+	for (size_t j = data_len; j; --j) {
+		crc ^= *data++;
+
+		for (size_t i = 8; i; --i) {
+			crc = (crc & 0x80) ? (crc << 1) ^ LC709203F_CRC_POLYNOMIAL : (crc << 1);
+		}
+	}
+
+	return crc;
+}
```

Comment:
Use zephyr crc8 API?

Reply:
Didn't know that zephyr has it, but somehow it was clear, that this already exists :smirk: